Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -273,12 +257,31 @@ private bool TryDeserializeField(FieldInfo field, IIonReader reader, IonType ion
             // for some Ion types need to be consumed in order to advance the reader.
             deserialized = ionSerializer.Deserialize(reader, field.FieldType, ionType);
             
-            if (options.IgnoreReadOnlyFields && field.IsInitOnly)
+            return !IgnoreDeserializedField(field, deserialized);
+        }
+
+        private bool IgnoreDeserializedProperty(object deserialized)
+        {
+            return options.IgnoreDefaults && deserialized == default;
+        }
+        
+        private bool IgnoreDeserializedField(FieldInfo field, object deserialized)
+        {
+            // Check if this field is really a backing field for a readonly property and
+            // if so, apply the ignore property logic instead of the ignore field logic.
+            if (!options.IgnoreReadOnlyProperties && IsBackingFieldForReadonlyProperty(field))
             {
-                return false;
+                return IgnoreDeserializedProperty(deserialized);
```

Comment:
Is this necessary? L278 has the same logic.

Reply:
`IgnoreDeserializedProperty` and `IgnoreDeserializedField` have similar but not identical logic.

Since this backing field is essentially a surrogate for a readonly property, we want to apply the property logic to this field (instead of the field logic).

Reply:
A backing field will have `field.IsInitOnly`= false because backing field only has `get`. It will go to L278 and it has the same conditional.

Reply:
A readonly property has a `{ get; }` accessor defined but the corresponding backing field will actually have `field.IsInitOnly = true` which could potentially result in this `IgnoreDeserializedField` method returning a false positive.

And even if you were right that the extra condition is harmless (at the moment), it's still worth separating out the IgnoreProperty and IgnoreField logic like this because we want this backing field to be treated the same as a property (however that logic is defined which could potentially change in the future).